### ISSUE TITLE
[ads] notes

### ISSUE THREAD

**Issue by youzi** [vigour-io/player#20]
**Ads**
Ad data is stored/written on the state (data) on each episode.

```js
// video state/data
{
  video: '{srcpath}',
  duration: 2000,
  progress: 0,
  ads: {
    0: [ '{vasturlA}', '{vasturlB}' ], // <==== shift when ad is finished
    50: [ '{vasturlC}', '{vasturlD}' ],
  },
  ad: { // <==== remove when ad is finished
    img: '{adthumburl}',
    title: '{adtitle}',
    url: '{clickthroughurl}',
    skipDelay: 10, // if number show skip counter, if 0 show skip button, if false show no skip info, 
    duration: 500,
    progress: 0
  }
}
```
![image](https://cloud.githubusercontent.com/assets/5662898/13812310/c7a6c492-eb7b-11e5-958a-d686252211b2.png)

![Uploading image.png…]()

Behaviour could be:
- When there is **no ads field** => player will read ad configuration and write a complete ad object to the state/data
- When there is an **ads field present** => player will use this object and remove the urls that it plays

- When a user has seen an adslot (ie preroll) for an episode he is not shown this slot again.
- At a certain interval this data needs to be 'refreshed' => re-enabling the removed adslots. This can happen by removing the `ads` field completely => forcing the player to write a new object.

*to be edited*

/cc @jimdebeer

**Comment by jimdebeer** [comment]
for the ads object

```
{
    0: [ 2321123 // timestamp means its watched ]
   '0.5': [ vastURL ] // from the prototype
}
```

idea is when there is a time stamp but time stamp is no longer valid (based on state.hub.clock -- else it overwritable, easy to hack) use delete on the nested val (so it inherits again) --- no remove! (since it sets the valiue to null and we want straight inheritance

_maybe store src of current ad -- but this is maybe for later_

**Comment by youzi** [comment]
```js
{
  player: { // subscribes to videodata
    video: {}
    controls: {
      $:true
    },
    adcontrols: {
      $:'ad'
    }
  }
}
```

**Comment by youzi** [comment]
Basic ads implemented in player@v2.0.0
